local.tf

```diff
@@ -19,8 +19,8 @@ locals {
   worker_group_count                 = length(var.worker_groups)
   worker_group_launch_template_count = length(var.worker_groups_launch_template)
 
-  default_ami_id_linux   = data.aws_ami.eks_worker.id
-  default_ami_id_windows = data.aws_ami.eks_worker_windows.id
+  default_ami_id_linux   = coalesce(local.workers_group_defaults.ami_id, data.aws_ami.eks_worker.id)
+  default_ami_id_windows = coalesce(local.workers_group_defaults.ami_id, data.aws_ami.eks_worker_windows.id)
```

Comment:
@ck3mp3r @max-rocket-internet I was just wondering how do we should handle hybrid kubernetes cluster (linux and windows). Do you think it worth to have `local.workers_group_defaults.ami_id` (default for linux) and `local.workers_group_defaults.ami_id_windows` (default for windows) to handle default ami id differently for both platforms ? Or we should let those people specify the ami_id for every worker group like https://github.com/terraform-aws-modules/terraform-aws-eks/issues/754 ?

Any thought ?

Reply:
@barryib good point, I think your suggestion works. If we specify our own defaults we might as well specify them per OS... isn't going to hurt and completes the pattern.

Reply:
@barryib this seems more involved than I anticipated 😆
How about we get this branch done first and then thing about how we could enhance in a next step?

Reply:
@max-rocket-internet any thought ?

Reply:
Sounds good to me.

Reply:
> handle default ami id differently for both platforms ?
Yep